src/net_processing.cpp

```diff
@@ -2175,6 +2175,10 @@ bool static ProcessMessage(CNode* pfrom, const std::string& strCommand, CDataStr
             nCMPCTBLOCKVersion = 1;
             connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::SENDCMPCT, fAnnounceUsingCMPCTBLOCK, nCMPCTBLOCKVersion));
         }
+
+        if (pfrom->nVersion >= WTXID_RELAY_VERSION) {
+            connman->PushMessage(pfrom, msgMaker.Make(NetMsgType::WTXIDRELAY));
```

Comment:
I wonder if this should be sent immediately in response to a VERSION message, rather than in response to a VERACK.  It looks like the way I've done it here creates a race condition where a peer could send a txid-based inv to us before it gets this message, which I think would cause relay of that transaction to fail.